### ISSUE TITLE
D3D9 profiling

### ISSUE THREAD

**Issue by iiv3** [apitrace/apitrace#461]
The `d3dretrace.exe` have a number of profiling options like `--pcpu`, `--pgpu` etc, but when used none of them produces any numerical output. The only output visible is the header line with comment containing the names the different fields.

Same applies to GUI `qapitrace.exe` profile option that calls `d3dretrace.exe` with a number of options and then display a result window that contains only static elements.

I tried the above under Win7. The binary is the msvc one from the download section, dated 2016-05-19 (the latest as of this moment).

If you don't consider this a bug, then please consider it a feature request. Implement at least `--pcpu`.

**Comment by jrfonseca** [comment]
Yes, profiling is not universally supported as described in http://apitrace.github.io/#features

I don't have much interest in this myself (I don't develop D3D9 apps nor do I see many people developing D3D9 apps these days), but I'm happy to accept patches.

I noticed that @EoD has some preliminary work done towards this in  https://github.com/EoD/apitrace/commits/d3d9-profiling-v2  It's described as an hack so I'm not sure how much work it is from being a polished change that could be merged.

**Comment by EoD** [comment]
I rebased (and partially reworked) @tizbac's work on d3d9 profiling. It is far from being polished enough to be merged. I created some executables from this work, if someone is interested in trying and maybe even polishing.

**Comment by iiv3** [comment]
There is a pull request from siro20 https://github.com/apitrace/apitrace/pull/476 that adds this feature to apitrace.

Unfortunately it has been sitting there for over 2 months without any comments.

Maybe you ignore it because one of the automated builds fails. However, I tried to look the reason for failure and the only thing I see is "Cancelled".

Please, take the time to review and test the changes.

**Comment by jrfonseca** [comment]
I didn't mean to ignore, but I've been incredible busy, professional and personally.  As a result there are a lot pending patches and issues to address.  I don't know when that will change.

See 75c7c00d8a825c16195ca4f8771abf402be65bfd

**Comment by siro20** [comment]
Can you have a look at the build server, please ?
It failed to build my PR #476  due to unrelated changes.

**Comment by jrfonseca** [comment]
Your branch probably needs to cherry-pick or rebase on top of a776411a5b92f645e8a6b49608dee81c7cd05994

**Comment by jrfonseca** [comment]
No nervermind. It can't be that...

**Comment by jrfonseca** [comment]
I believe this is the error I fixed in 9022a435da9269710a5056a9161c65c901e6cac7.  Anyway, if you rebase on top of master it should go away.

**Comment by iiv3** [comment]
Would You Kindly merge @siro20's Push Request for this feature?

Miraculously, it is still marked with green light.
